packages/babel-plugin-proposal-class-properties/test/fixtures/private/logical-assignment/output.js

```diff
@@ -0,0 +1,35 @@
+function _classPrivateFieldSet(receiver, privateMap, value) { var descriptor = privateMap.get(receiver); if (!descriptor) { throw new TypeError("attempted to set private field on non-instance"); } if (descriptor.set) { descriptor.set.call(receiver, value); } else { if (!descriptor.writable) { throw new TypeError("attempted to set read only private field"); } descriptor.value = value; } return value; }
+
+function _classPrivateFieldGet(receiver, privateMap) { var descriptor = privateMap.get(receiver); if (!descriptor) { throw new TypeError("attempted to get private field on non-instance"); } if (descriptor.get) { return descriptor.get.call(receiver); } return descriptor.value; }
+
+var _nullish = new WeakMap();
+
+var _and = new WeakMap();
+
+var _or = new WeakMap();
+
+class Foo {
+  constructor() {
+    _nullish.set(this, {
+      writable: true,
+      value: 0
+    });
+
+    _and.set(this, {
+      writable: true,
+      value: 0
+    });
+
+    _or.set(this, {
+      writable: true,
+      value: 0
+    });
+  }
+
+  test() {
+    _classPrivateFieldGet(this, _nullish) ?? _classPrivateFieldSet(this, _nullish, 42);
```

Comment:
We still need to add a test case for deep property access (`this.self().#foo ??= bar`). [repl](https://babeljs.io/repl/build/23799/#?browsers=chrome%2080&build=&builtIns=false&spec=false&loose=false&code_lz=MYGwhgzhAEBiD29oG8BQ1oGIB2BXEIAlhABbQC80ADANzrT0QCmIAZgBQCUK9GATkwAuuPtmiCSxOhgC-qeoKYRBXHhgwTiAOmZsuWnPiKloAflOUALACZp0OTKA&debug=false&forceAllTransforms=false&shippedProposals=true&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=stage-3&prettier=false&targets=&version=7.10.2%2Bpr.11702&externalPlugins=) That shows us we still have one last bug, because the memoization isn't being done on the correct `this.self()` access.